packages/telemetry/src/context-aware-slog.js

```diff
@@ -0,0 +1,475 @@
+        assert(!!crankContext);
+        crankContext['crank.vatID'] = finalBody.vatID;
+        break;
+      }
+      case SLOG_TYPES.CRANK.START: {
+        crankContext = {
+          'crank.num': finalBody.crankNum,
+          'crank.type': finalBody.crankType,
+        };
+        break;
+      }
+      case SLOG_TYPES.DELIVER: {
+        if (replayContext) {
+          assert(finalBody.replay);
+          replayContext = {
+            ...replayContext,
+            'crank.deliveryNum': finalBody.deliveryNum,
+            'crank.vatID': finalBody.vatID,
+          };
+        } else {
+          assert(!!crankContext);
+          crankContext = {
+            ...crankContext,
+            'crank.deliveryNum': finalBody.deliveryNum,
+            'crank.vatID': finalBody.vatID,
+          };
+        }
+
+        delete finalBody.deliveryNum;
+        delete finalBody.replay;
+
+        break;
+      }
+      case SLOG_TYPES.DELIVER_RESULT: {
+        delete finalBody.deliveryNum;
+        delete finalBody.replay;
+
+        break;
+      }
+      case SLOG_TYPES.KERNEL.INIT.START: {
+        initContext = { init: true };
+        break;
+      }
+      case SLOG_TYPES.REPLAY.START: {
+        replayContext = { replay: true };
+        break;
+      }
+      // eslint-disable-next-line no-restricted-syntax
+      case SLOG_TYPES.RUN.START: {
+        if (!(triggerContext || finalBody.runNum === 0)) {
+          const blockTime = finalBody.blockTime || blockContext?.['block.time'];
+
+          assert(blockTime);
+          // TBD: add explicit slog events of both timer poll and install bundle
+          triggerContext = {
+            'run.id': `timer-${finalBody.blockHeight}`,
+            'run.trigger.time': blockTime,
+            'run.trigger.type': 'timer',
+          };
+          persistContext(triggerContext);
+        }
+
+        if (!triggerContext) triggerContext = {};
+        triggerContext = {
+          ...triggerContext,
+          'run.num': `${finalBody.runNum}`,
+        };
+
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.BEGIN_BLOCK: {
+        blockContext = {
+          'block.height': finalBody.blockHeight,
+          'block.time': finalBody.blockTime,
+        };
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.BOOTSTRAP_BLOCK.START: {
+        blockContext = {
+          'block.height': finalBody.blockHeight || 0,
+          'block.time': finalBody.blockTime,
+        };
+        triggerContext = {
+          'run.id': `bootstrap-${finalBody.blockTime}`,
+          'run.trigger.time': finalBody.blockTime,
+          'run.trigger.type': 'bootstrap',
+        };
+        break;
+      }
+      case SLOG_TYPES.SWINGSET.BRIDGE_INBOUND:
+      case SLOG_TYPES.SWINGSET.DELIVER_INBOUND: {
+        const [blockHeight, txHash, msgIdx] = (
+          finalBody.inboundNum || ''
+        ).split('-');
+        const triggerType = body.type.split('-')[2];
+
+        triggerContext = {
```

Comment:
Please sort these as they were described in the issue. If the otel collector and log UI is any good it'll keep the same order, allowing us to more quickly see together related attributes.

And let's start with a `run.num` placeholder so that when it gets added it appears first

```suggestion
        triggerContext = {
          'run.num': null,
```

Reply:
Will do this. But I heavily doubt keys order will be honoured.

Reply:
At the very least they would be by a "slog transformer" that simply serializes back to a file